yroots/subdivision.py

```diff
@@ -620,76 +632,76 @@ def subdivision_solve_nd(funcs,a,b,deg,target_deg,interval_data,root_tracker,tol
             for new_a, new_b in intervals:
                 subdivision_solve_nd(funcs,new_a,new_b,deg,target_deg,interval_data,root_tracker,tols,max_level,level=level+1, method=method)
             return
         else:
-            #if the function changes sign on at least one subinterval, skip the checks
+            # if the function changes sign on at least one subinterval, skip the checks
             if np.any(change_sign):
                 cheb_approx_list.append(coeff)
                 continue
-            #Run checks to try and throw out the interval
+            # Run checks to try and throw out the interval
             if interval_data.check_interval(coeff, approx_error, a, b):
                 return
 
             cheb_approx_list.append(coeff)
 
-    #Make the system stable to solve
+    # Make the system stable to solve
     coeffs, good_approx, approx_errors = trim_coeffs(cheb_approx_list, tols.abs_approx_tol, tols.rel_approx_tol, inf_norms, approx_errors)
 
-    #Used if subdividing further.
+    # Used if subdividing further.
     good_degs = [coeff.shape[0] - 1 for coeff in coeffs]
     good_zeros_tol = max(tols.min_good_zeros_tol, np.sum(np.abs(approx_errors))*tols.good_zeros_factor)
+        
+    # Runs the same things as above, but we want to get rid of that eventually so keep them seperate.
+    if np.any(np.array([coeff.shape[0] for coeff in coeffs]) > target_deg) or not good_approx:
+        intervals = get_subintervals(a,b,np.arange(dim),interval_data,cheb_approx_list,change_sign,approx_errors,True)
+        for new_a, new_b in intervals:
+            subdivision_solve_nd(funcs,new_a,new_b,deg, target_deg,interval_data,root_tracker,tols,max_level,good_degs,level+1, method=method, use_target_tol=True)
 
-    #Check if everything is linear
-    if np.all(np.array([coeff.shape[0] for coeff in coeffs]) == 2):
+    # Check if any approx error is greater than target_tol for Macaulay method
+    elif np.any(np.array(approx_errors) > np.array(tols.target_tol) + tols.rel_approx_tol*np.array(inf_norms)):
+        intervals = get_subintervals(a,b,np.arange(dim),interval_data,cheb_approx_list,change_sign,approx_errors,True)
+        for new_a, new_b in intervals:
+            subdivision_solve_nd(funcs,new_a,new_b,deg, target_deg,interval_data,root_tracker,tols,max_level,good_degs,level+1, method=method, use_target_tol=True)
+
+    # Check if everything is linear
+    elif np.all(np.array([coeff.shape[0] for coeff in coeffs]) == 2):
         if deg != 2:
-            subdivision_solve_nd(funcs,a,b,2,target_deg,interval_data,root_tracker,tols,max_level,good_degs,level, method=method)
+            subdivision_solve_nd(funcs,a,b,2,target_deg,interval_data,root_tracker,tols,max_level,good_degs,level, method=method, use_target_tol=True)
             return
         zero, cond = solve_linear(coeffs)
-        grad = [MultiCheb(c).grad(zero) for c in coeffs]
```

Comment:
Oh yeah, this was just for testing purposes so I guess we don't need this anymore